src/v/features/feature_table_snapshot.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "feature_table_snapshot.h"
+
+#include "feature_table.h"
+
+namespace features {
+
+feature_table_snapshot feature_table_snapshot::from(const feature_table& ft) {
+    feature_table_snapshot fts;
+
+    fts.states.reserve(feature_schema.size());
+
+    fts.version = ft.get_active_version();
+    fts.license = ft.get_license();
+    for (const auto& state : ft._feature_state) {
+        auto& name = state.spec.name;
+        fts.states.push_back(feature_state_snapshot{
+          .name = ss::sstring(name), .state = state._state});
+    }
+
+    return fts;
+}
+
+void feature_table_snapshot::apply(feature_table& ft) const {
+    ft.set_active_version(version);
+    ft._license = license;
+    for (auto& snap_state : ft._feature_state) {
+        auto snap_state_iter = std::find_if(
+          states.begin(),
+          states.end(),
+          [&snap_state](feature_state_snapshot const& s) {
+              return s.name == snap_state.spec.name;
```

Comment:
Any reason to use the name and not enum? Especially if we're not logging anything

Reply:
The `feature` enum contains bitmasks, which at some point we'll probably want to change when we run out of bits, so the serialization format uses string names (controller log entries also use the string name of features)